src/components/PostDetails.tsx

```diff
@@ -1,100 +1,34 @@
-import React, { useEffect, useState } from 'react';
+import { FC, useEffect, useState, useCallback } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { fetchComments, removeComment } from '../features/comments';
 
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+export const PostDetails: FC = () => {
+  const { comments, loaded, hasError } = useAppSelector(
```

Comment:
It's better to destructure the state in the selector to make the code more readable and maintainable. You can create separate selectors for comments, loaded, and hasError in the comments slice and use them here.